### PR TITLE
Update Integration Catalog docs to reflect recent changes

### PR DESCRIPTION
**Pull request by ShaunEvening**
Issue:

## What I did
- Rename addon-catalog doc to integration-catalog
- Add section on recipes

## How to test

- [ ] Is this testable with Jest or Chromatic screenshots?
- [ ] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

File: docs/integrations/integration-catalog.md [hunk 1]
```diff
@@ -1,8 +1,12 @@
 ---
-title: 'Add to the addon catalog'
+title: 'Add to the Integration catalog'
```
Comment (kylegach) [inline docs/integrations/integration-catalog.md:2]: We typically use "Sentence case" for our headings. Though I can see a case for capitalizing key terms like this. Can you explain your reasoning here?

File: docs/integrations/integration-catalog.md [hunk 2]
```diff
@@ -1,8 +1,12 @@
...
 ---
 
-Storybook addons are listed in the [catalog](https://storybook.js.org/addons/) and distributed via npm. The catalog is populated by querying npm's registry for Storybook-specific metadata in `package.json`.
+Storybook has two types of integrations, addons and recipes, and are listed in the [integration catalog](https://storybook.js.org/integrations/).
```
Comment (kylegach) [inline docs/integrations/integration-catalog.md:5]:
```suggestion
Storybook has two types of integrations, addons and recipes, which are listed in the [integration catalog](https://storybook.js.org/integrations/).
```

File: docs/integrations/integration-catalog.md [hunk 3]
```diff
@@ -85,3 +89,17 @@ The `package.json` above appears like below in the catalog. See an example of a
...
+
+## Recipes
+
+Recipes are a set of instructions to integrate your favorite libraries into Storybook in cases where an addon does not exist or the integration requires some manual effort.
+
+### Who owns them?
+
+Recipes are written and maintained by the Storybook team. We create recipes based on community popularity, tool maturity, and stability of the integration. Our goal is to ensure that recipes continue to work over time.
+
+Not finding the recipe that you want? If it's popular in the community, our docs team will write one. In the mean time, try googling for a solution — it's likely that someone has the same requirements as you do. You can also help us out by writing recipes on your own site which speeds up the research process.
```
Comment (kylegach) [inline docs/integrations/integration-catalog.md:101]:
```suggestion
Not finding the recipe that you want? If it's popular in the community, our docs team will write one. In the mean time, try searching for a solution — it's likely that someone has the same requirements as you do. You can also help us out by writing recipes on your own site which speeds up the research process.
```

File: docs/integrations/integration-catalog.md [hunk 4]
```diff
@@ -85,3 +89,17 @@ The `package.json` above appears like below in the catalog. See an example of a
...
+
+### Request a recipe
+
+If you'd like to request a recipe, head over to the [#maintenance channel](https://discord.com/channels/486522875931656193/490070912448724992) of our community discord and ask
```
Comment (kylegach) [inline docs/integrations/integration-catalog.md:105]:
```suggestion
If you'd like to request a recipe, head over to the [#maintenance channel](https://discord.com/channels/486522875931656193/490070912448724992) of our community Discord and ask.
```

**Review by kylegach — APPROVED** [review]
Nicely re-structured!

**Review by domyen — COMMENTED** [review]
Agree with KGs suggestions, otherwise lgtm

**Comment by kylegach** [comment]
FYI, I applied the `patch` label so that these changes will be patched back into `main` as part of the next release.

**Comment by kylegach** [comment]
And now I've _removed_ the `patch` label in favor of handling this with a PR targeting `main`: #20192

This is because the corresponding frontpage PR, https://github.com/storybookjs/frontpage/pull/442, links to `/addons/integration-catalog` and will do so across versions. So that URL needs to work in both `next` (this PR) and "latest" (`#20192`).

**Pull request merged**